lib/elixir/lib/supervisor/spec.ex

```diff
@@ -75,6 +75,10 @@ defmodule Supervisor.Spec do
     * `:transient` - the child process is restarted only if it
       terminates abnormally, i.e., with an exit reason other than
       `:normal`, `:shutdown` or `{:shutdown, term}`
+  
+  Notice that whenever supervisor with `:transient` restart value reaches `max_restart_intensity`,
+  it would not be restarted, since exit reason in this case is `:shutdown` and it is not considered 
```

Comment:
I think that this is a good thing to mention, because it's not obvious that following pure "let it fail" principle in some cases will bring you to the zombie Erlang VMs that doesn't do anything.

Our case is here: [distillery/#118](https://github.com/bitwalker/distillery/issues/118).